src/Microsoft.DotNet.Wpf/src/Shared/MS/Win32/SafeNativeMethodsOther.cs

```diff
@@ -124,23 +124,24 @@ public static int GetCaretBlinkTime()
         public const UInt16 C3_IDEOGRAPH  = 0x0100;
         public const UInt16 C3_KASHIDA    = 0x0200;
 
-        public static bool GetStringTypeEx(uint locale, uint infoType, char[] sourceString, int count,
-            UInt16[] charTypes)
+        public static unsafe bool GetStringTypeEx(uint locale, uint infoType, ReadOnlySpan<char> sourceString, ReadOnlySpan<UInt16> charTypes)
```

Comment:
`ReadOnlySpan<UInt16> charTypes` here is not right and could potentially cause problems. `charTypes` is modified by the call to GetStringTypeEx.
```suggestion
        public static unsafe bool GetStringTypeEx(uint locale, uint infoType, ReadOnlySpan<char> sourceString, Span<UInt16> charTypes)
```

Reply:
I mean it won't cause any problems until you start feeding it wrong stuff, which means you don't know what you're doing with the function but I get your point, addressed.